### PR TITLE
dev-cmd/contributions: Count the number of commits a user committed

### PR DESCRIPTION
**Pull request by issyl0**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

- Part of https://github.com/Homebrew/brew/issues/13642.
- The GitHub list commits API now supports this filtering (https://docs.github.com/en/rest/commits/commits?apiVersion=2022-11-28#list-commits--parameters), because I wrote it. :-)
- Authoring a commit and committing a commit are two separate concepts: author is the person who wrote the code and, in old parlance, the committer is the person who applied the patch (remember when we sent patches to mailing lists?).
- In practice for us in Homebrew, this occurs when we make a change in GitHub's web editor, or, more obviously, when BrewTestBot pushes `homebrew-core` commits from users (then, `BrewTestBot` is the `committer`).

TODO:

- [x] Don't double count contributions for which the author and committer are the same. ~(See the following example for why this would be bad.)~

Example:

```
❯ brew contributions --repos=primary --user=issyl0 --csv
issyl0 contributed 1626 times in all time.
user,repo,author,committer,coauthorships,reviews,total
issyl0,brew,392,147,13,142,694
issyl0,core,473,73,24,357,927
issyl0,cask,4,0,0,1,5
issyl0,all,869,220,37,500,1626
```

### CODE REVIEW

File: Library/Homebrew/dev-cmd/contributions.rb
Comment (nandahkrishna) [inline Library/Homebrew/dev-cmd/contributions.rb:179]: Am I right in understanding that:
| Author       | Committer    | Counted In |
|--------------|--------------|------------|
| `user`         | other or `user` | `commits_authored`     |
| other or `user` | `user`        | `commits_committed`  |

are all the possibilities? So `commits_committed = commits_authored - commits_committed` doesn't seem correct, right? For example, let's say I have exactly 5 commits in homebrew-core:
* Author – `nandahkrishna`; Committer – `BrewTestBot`
* Author – `nandahkrishna`; Committer – `nandahkrishna`
* Author – `nandahkrishna`; Committer – `nandahkrishna`
* Author – `samford`; Committer – `nandahkrishna`
* Author – `issyl0`; Committer – `nandahkrishna`

We would finally have `commits_authored = 3` and `commits_committed = -1` (originally `4`, but `commits_committed = 3 - 4 = -1`). There aren't too many commits of the last kind especially after auto-merge was introduced though.

Comment (issyl0) [reply Library/Homebrew/dev-cmd/contributions.rb:179]: Maybe my 10pm maths was flawed too. I did take my high school maths exam three times. 🙈 I'll work all this out again.

**Review by MikeMcQuaid — APPROVED** [review]

File: Library/Homebrew/dev-cmd/contributions.rb
Comment (issyl0) [reply Library/Homebrew/dev-cmd/contributions.rb:179]: What I was trying to get to was...

Assume I have 4 commits in Cask:

| Author | Committer |
| issyl0 | BrewTestBot |
| issyl0 | reitermarkus |
| issyl0 | commitay |
| issyl0 | BrewTestBot |

`commits_authored = 4`, `commits_committed = 0`, so `commits_committed` was coming out as 4 (4 - 0 = 4) before the `.positive?` check (`> 0`).

I can see now how this would be flawed for your numbers. Maybe this needs some tests with a variety of different data. 😅

I guess we could do this the "proper" way by getting the SHAs of the authored commits and the SHAs of the committed commits and getting the uniques in each and counting _those_ to deduplicate. But that seemed like a lot.

Comment (issyl0) [reply Library/Homebrew/dev-cmd/contributions.rb]: [693d609](https://github.com/Homebrew/brew/pull/14860/commits/693d609730e5f20c0347000850583506c825abae) should fix this, but please check my maths again!

**Review by MikeMcQuaid — APPROVED** [review]

**Pull request merged**
